File: src/ruby/lib/dependabot/job/models/group_pull_request.rb
# typed: strict
# frozen_string_literal: true

require 'sorbet-runtime'
require_relative 'pull_request_dependency'

module Dependabot
  module Job
    module Models
      # Represents an existing group pull request.
      class GroupPullRequest < T::Struct
        extend T::Sig

        # The name of the dependency group.
        const :dependency_group_name, T.nilable(String), default: nil

        # The dependencies included in this group pull request.
        const :dependencies, T::Array[PullRequestDependency], default: []

        # The pull request number.
        const :number, T.nilable(Integer), default: nil

        # The pull request URL.
        const :url, T.nilable(String), default: nil

        sig { params(hash: T::Hash[Symbol, T.untyped]).returns(GroupPullRequest) }
        def self.from_hash(hash)
          dependencies = Array(hash[:dependencies] || []).map do |dep_hash|
            PullRequestDependency.from_hash(dep_hash)
          end

          new(
            dependency_group_name: hash[:dependency_group_name],
            dependencies: dependencies,
            number: hash[:number],
            url: hash[:url]
          )
        end

        sig { returns(T::Hash[Symbol, T.untyped]) }
        def to_hash
          {
            dependency_group_name: dependency_group_name,
            dependencies: dependencies.map(&:to_hash),
            number: number,
            url: url
          }.compact
        end
      end
    end
  end
end

File: src/dotnet/src/Dependabot.Job/Models/GroupPullRequest.cs
using System.Collections.Immutable;

namespace Dependabot.Job;

/// <summary>
/// Represents an existing group pull request.
/// </summary>
public sealed record GroupPullRequest
{
  /// <summary>
  /// Gets or sets the name of the dependency group.
  /// </summary>
  public string? DependencyGroupName { get; init; }

  /// <summary>
  /// Gets or sets the dependencies included in this group pull request.
  /// </summary>
  public ImmutableArray<PullRequestDependency> Dependencies { get; init; } = [];

  /// <summary>
  /// Gets or sets the pull request number.
  /// </summary>
  public int? Number { get; init; }

  /// <summary>
  /// Gets or sets the pull request URL.
  /// </summary>
  public string? Url { get; init; }
}
